sip-core/src/main/java/de/ikor/sip/foundation/core/declarative/DeclarationsRegistry.java

```diff
@@ -57,11 +60,111 @@ public DeclarationsRegistry(
+            base.getRequestMapper()
+                .ifPresent(
+                    mapper -> {
+                      if (isRequestMappingOverridden(connectorDefinition, mapper)) {
+                        throw SIPFrameworkInitializationException.init(
+                            "Request mapping in connector '%s' is defined, but overridden",
+                            connectorDefinition.getId());
+                      }
+                    });
+            base.getResponseMapper()
+                .ifPresent(
+                    mapper -> {
+                      if (isResponseMappingOverridden(connectorDefinition, mapper)) {
+                        throw SIPFrameworkInitializationException.init(
+                            "Response mapping in connector '%s' is defined, but overridden",
+                            connectorDefinition.getId());
+                      }
+                    });
+          }
+        });
+  }
+
+  private boolean isRequestMappingOverridden(
+      ConnectorDefinition connectorDefinition,
+      RequestMappingRouteTransformer<Object, Object> mapper) {
+    return connectorDefinition.getOrchestrator()
+            instanceof ConnectorOrchestrator connectorOrchestrator
+        && (!mapper.equals(connectorOrchestrator.getRequestRouteTransformer()));
+  }
+
+  private boolean isResponseMappingOverridden(
+      ConnectorDefinition connectorDefinition,
+      ResponseMappingRouteTransformer<Object, Object> mapper) {
+    return connectorDefinition.getOrchestrator()
+            instanceof ConnectorOrchestrator connectorOrchestrator
+        && (!mapper.equals(connectorOrchestrator.getResponseRouteTransformer()));
+  }
+
+  private void checkForMissingScenarioParent() {
+    applicationContext
+        .getBeansWithAnnotation(IntegrationScenario.class)
+        .values()
+        .forEach(
+            o -> {
+              if (!(o instanceof IntegrationScenarioBase)) {
+                throw SIPFrameworkInitializationException.init(
+                    "Annotated IntegrationScenario %s is missing IntegrationScenarioBase parent class.",
+                    o.getClass().getName());
+              }
+            });
+  }
+
+  private void checkForMissingConnectorGroupParent() {
+    applicationContext
+        .getBeansWithAnnotation(ConnectorGroup.class)
+        .values()
+        .forEach(
+            o -> {
+              if (!(o instanceof ConnectorGroupBase)) {
+                throw SIPFrameworkInitializationException.init(
+                    "Annotated ConnectorGroup %s is missing ConnectorGroupBase parent class.",
+                    o.getClass().getName());
+              }
+            });
+  }
+
+  private void checkForMissingConnectorParent() {
+    applicationContext
+        .getBeansWithAnnotation(InboundConnector.class)
+        .values()
+        .forEach(
+            o -> {
+              if (!(o instanceof InboundConnectorBase)) {
```

Comment:
Please double check if this works for Rest as well.

Reply:
Both rest and generic inbound connectors have InboundConnectorBase as parent